### PR TITLE
WIP & Discussion: docs structure

### PR DESCRIPTION
**Pull request by phryneas**
What I did here was to split up the `createApi` API doc return value page into a whole own section. We need to document all these methods individually.

I've already talked to @msutkowski  and he'll take over from here (:trophy:) while I dig into the types for the other PRs :smile:  

@markerikson Any feedback on this would be highly appreciated :)

### CODE REVIEW

**Comment by markerikson** [comment]
I reworked the "API Slice Contents" area to be one reference page per category ("Redux Integration", "Endpoints", etc. No need to have a separate page for every individual function. 

Also merged down from `next` to pick up the latest changes, and wrote a ton of actual content for those pages .

File: docs/api/created-api/hooks.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  skip?: boolean;
+  refetchOnMountOrArgChange?: boolean | number;
+  selectFromResult?: QueryStateSelector;
+};
+
+type UseQueryResult<T> = {
+  // Base query state
+  status: 'uninitialized' | 'pending' | 'fulfilled' | 'rejected'; // @deprecated - A string describing the query state
+  originalArgs?: unknown; // Arguments passed to the query
+  internalQueryArgs?: unknown;
```
Comment (Shrugsy) [inline docs/api/created-api/hooks.md:60]:
```suggestion
```
Outdated property

File: docs/api/created-api/hooks.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  data?: T; // Returned result if present
+  error?: unknown; // Error result if present
+  requestId?: string; // A string generated by RTK Query
+  endpoint?: string; // The name of the given endpoint for the query
```
Comment (Shrugsy) [inline docs/api/created-api/hooks.md:64]:
```suggestion
  endpointName?: string; // The name of the given endpoint for the query
```

Outdated property name

File: docs/api/created-api/hooks.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  startedTimeStamp?: number; // Timestamp for when the query was initiatied
```
Comment (Shrugsy) [inline docs/api/created-api/hooks.md:65]:
```suggestion
  startedTimeStamp?: number; // Timestamp for when the query was initiated
```

Fixes a typo

**Pull request merged**